torch_xla/experimental/fori_loop.py

```diff
@@ -10,66 +10,95 @@
 from torch._ops import HigherOrderOperator
 import torch._higher_order_ops.while_loop
 from torch._higher_order_ops.while_loop import while_loop_op
+from torch._higher_order_ops.while_loop import while_loop as torch_while_loop
 
 
-def fori_loop(lower, upper, user_body_func, *init_val):
+# TODO(@manfei): treat *input_value
+def fori_loop(upper, lower, body_fun, init_val, input_value):
 
   device = xm.xla_device()
 
-  def cond_fn(upper, lower, *init_val):
-    return lower[0] < upper[0]
+  one_value = torch.tensor([1], dtype=torch.int32, device=device)
 
-  def body_fn(upper, lower, *init_val):
-    one_value_i = torch.ones(1, dtype=torch.int32, device=device)
-    res_list = list(user_body_func(*init_val))
-    res_list.insert(0, lower)
-    res_list.insert(0, torch.sub(upper, one_value_i))
-    return res_list
+  if (hasattr(body_fun, 'weight') or hasattr(body_fun, 'bias')):
```

Comment:
what is the plan to remove this condition?

Reply:
actually why do we special casing the weight and bias? Is it only for the linear layer?

Reply:
yes, its only for the linear layer, and plan to remove this condition is: only check whether `additional_inputs`(weight/bias) exist for code of weight/bias in the next step